pennylane/_qubit_device.py

```diff
@@ -31,20 +31,22 @@
 class QubitDevice(Device):
     """Abstract base class for PennyLane qubit devices.
 
-    The following abstract methods **must** be defined:
-
-    * :meth:`~.probability`: returns the probability or marginal probability from the
-      device after circuit execution. :meth:`~.marginal_prob` may be used here.
+    The following abstract method **must** be defined:
 
     * :meth:`~.apply`: append circuit operations, compile the circuit (if applicable),
       and perform the quantum computation.
 
-    Where relevant, devices that generate their own samples (such as hardware) should
-    overwrite the following methods:
+    Devices that generate their own samples (such as hardware) **must**
+    overwrite the following method:
 
     * :meth:`~.generate_samples`: Generate samples from the device from the
       exact or approximate probability distribution.
```

Comment:
Worth also mentioning that devices that generate their own samples may optionally override `probability`. If it's not overridden, PennyLane will automatically compute the probabilities from the generated samples.

For example, this affects the Qiskit plugin --- Qiskit is able to return probabilities _and_ samples, so it would be redundant to have the `QubitDevice` compute the probabilities again.